packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -2165,7 +2182,7 @@ class _CupertinoTimerPickerState extends State<CupertinoTimerPicker> {
 
   Widget _buildHourPicker(EdgeInsetsDirectional additionalPadding, Widget selectionOverlay) {
     return CupertinoPicker(
-      scrollController: FixedExtentScrollController(initialItem: selectedHour!),
+      scrollController:hourController ,
```

Comment:
selectedHour may change during life of the widget

It seems we need to do something like `jumpTo` or `jumpToItem` here to pick up the change.
It relates to other `selected...` too.

And, we need a test that will catch the problem, because current test coverage does not catch such cases.

Reply:
Fixed in other PR https://github.com/flutter/flutter/pull/135355
so reverting my commit for this file.